core/src/main/csharp/ch/cyberduck/core/diagnostics/TcpReachability.cs

```diff
@@ -81,31 +80,34 @@ public bool isReachable(Host h)
                             // not an exception?
                             case WebExceptionStatus.Success:
                                 return true;
-                        }
 
-                        return false;
+                            default:
+                                if (Log.isDebugEnabled())
+                                {
+                                    Log.debug($"WebException thrown with status {e.Status} for {h}");
+                                }
+
+                                return false;
+                        }
                     }
-                    catch (Exception e)
+                }
+                else
+                {
+                    if (Log.isDebugEnabled())
                     {
-                        Log.error($"Generic exception while checking for reachability for {h}", e);
-                        return false;
+                        Log.debug($"Try TCP connection to {h.getHostname()}:{h.getPort()}");
                     }
-                default:
-                    try
-                    {
-                        if (Log.isDebugEnabled())
-                        {
-                            Log.debug($"Try TCP connection to {h.getHostname()}:{h.getPort()}");
-                        }
 
-                        TcpClient c = new TcpClient(h.getHostname(), h.getPort());
-                        c.Close();
-                        return true;
-                    }
-                    catch (SocketException e)
+                    using (new TcpClient(h.getHostname(), h.getPort()))
                     {
-                        return false;
+                        return true;
                     }
+                }
+            }
+            catch (Exception e)
+            {
+                Log.warn($"Reachability check for {h} failed {e.Message}");
```

Comment:
One large catch around all of the networking stuff.
No need to have two catch blocks with exactly the same content (or rather: no content. Up until now reachability results for Tcp-tests didn't end up in the log)